.pre-commit-hooks.yaml

```diff
@@ -4,3 +4,10 @@
     entry: gitlint
     args: [--staged, --msg-filename]
     stages: [commit-msg]
+-   id: gitlint0
```

Comment:
FWIW I've been using the "write each message to file, pass to pre-commit gitlint one at a time" approach for a while now. It works, but it's somewhat messy in the first place, and made even more so by pull requests and pushes in GitHub having different ways to get the commit messages in them:

https://github.com/scop/pytekukko/blob/d101b221501ccc9514c0f2c95eb7044079443291/.github/workflows/check.yml#L49

Maybe there's a cleaner way to get the commits and their messages via plain git, but I didn't have success with that for some reason when I last looked into it and didn't feel like spending too much time on it. IIRC the problems I ran into were primarily with pull requests, but the details escape me at the moment.

If gitlint accepted a single msg-filename with multiple messages in it (maybe just the usual git log output) and someone figured out the git command to get the log for commits included in a PR, that'd provide some cleanup opportunities over the above implementation.

Reply:
Combining the above inputs, I guess we're looking at something like this:
```yaml
-   id: gitlint-ci
    name: gitlint
    language: python
    additional_dependencies: ["./gitlint-core[trusted-deps]"]
    entry: gitlint
    always_run: true
    pass_filenames: false
    stages: [manual]
```
This hook would allow us to invoke gitlint on the CI with `pre-commit run --hook-stage manual gitlint-ci`. Just `HEAD` would be linted by default, but it is possible to specify a range of commits using `args: ["--commits", "commit..range"]`. AFAIK there is no easier way to pass arguments to pre-commit hooks in order to customize them at will.

Reply:
@scop @KSmanis Thanks a lot for your feedback and proposals. I share scop's opinion not only in Github but in Gerrit too. It is quite messy but at least in Gerrit+Jenkins we do not need any range. Each commit is evaluated separately by the CI and HEAD~1 (gitlint default behavior) is enough.
I like KSmanis adaptations and I pushed them in a new personal branch at https://github.com/guillaumelambert/gitlint/tree/gitlint-ci.
Fair enough to use manual stage as suggested above even I still think it is a pity that there would no commit stage proposed yet.

@sigmavirus24  I've modified the PR title to reflect our talks. Should I create a new PR or is it possible to reopen this PR? This way, we will keep a trace of all remarks at the same place.

Reply:
You can reopen this one, I think

Reply:
@sigmavirus24 Thanks. I cannot do it by myself since only repo committers and maintainers are allowed to do this. Can you take care of it ?

Reply:
@sigmavirus thanks for reopening the PR. I've pushed the update. (I would have liked to change my source branch but surprisingly this does not appear to be supported by github so I had to stick with the old branch naming that is a bit confusing now...). Hope this helps.

Reply:
Hello

I didn't receive any feedback since the last PR modification.
And I think it covers all or almost all previous feedback received.
Just let me know if there is still anything blocking its approval.

Kind Regards
Guillaume

Reply:
It seems  a few commits  were integrated in the main branch meanwhile. So I rebased this.
@sigmavirus24 @jorisroovers Just let me know if there is something more I can do.